Make SBBreakpoint/SBBreakpointLocation.SetCondition(nullptr) work again.

jimingham opened the pull request:
The addition of the StopCondition in the lldb_private layer meant that clearing a breakpoint condition with:

sb_break.SetCondition(nullptr);

now crashes.  Also, GetCondition for an empty condition used to return a nullptr, but now it returns "".

This patch fixes that crash and makes the SB GetCondition always return nullptr for an empty condition.

jimingham committed:
Make SBBreakpoint/SBBreakpointLocation.SetCondition(nullptr) work again.

The addition of the StopCondition in the lldb_private layer meant that
clearing a breakpoint condition with:

sb_break.SetCondition(nullptr);

now crashes.  Also, GetCondition for an empty condition used to return
a nullptr, but now it returns "".

This patch fixes that crash and makes the SB GetCondition always return
nullptr for an empty condition.

jimingham committed:
Move the unittest to the API directory and make it a pure SB test.

jimingham committed:
formatting

JDevlieghere commented on lldb/unittests/Breakpoint/CMakeLists.txt:
```diff
@@ -1,10 +1,15 @@
   LINK_COMPONENTS
     Support
   LINK_LIBS
+    liblldb
```
When you mix static and dynamic libraries that contain the same code, you end up with two copies of the same symbols which leads to various hard-to-debug issues (like having two copies of globals). When you say it "works" you mean you don't notice any of these issues for this particular test, but that doesn't make it any less wrong :-) 

I like the idea of enforcing this at the CMake layer. I've filed: https://github.com/llvm/llvm-project/issues/162378

jimingham commented:
Grrr... This is failing the Linux test bot because someone had previously changed an API in Stream.h but didn't update the documentation and for some reason the test I added included that in a way that caused a `-Wdocumentation` warning, which caused the Linux bot - which runs `-Werror` - to fail.
I fixed that in:

https://github.com/llvm/llvm-project/pull/162401

jimingham committed:
Remove an unneeded header that ended up including the lldb private headers.

This works around the -Wdocumentation error when building the test.

jimingham committed:
Update lldb/unittests/API/SBBreakpointClearConditionTest.cpp

Use the LLDB.h so the test can test documentation as a side-effect.

Co-authored-by: Jonas Devlieghere <jonas@devlieghere.com>

jimingham committed:
Update lldb/unittests/API/SBBreakpointClearConditionTest.cpp

Correct the dash pattern.

Co-authored-by: Jonas Devlieghere <jonas@devlieghere.com>

jimingham committed:
Repair a bad merge from accepting a GitHub suggested change
in the GitHub UI.
